# Fix incompatible function pointer types passing void
# Patch was submitted by user  nvinson234 on gentoo bugzilla
#
# A fix is being discussed upstream at https://gitlab.gnome.org/GNOME/vala/-/issues/1413
# Until then, this patch should allow building libsecret with clang-16 (and newer)
# Bug: https://bugs.gentoo.org/894328
--- a/libsecret/test-vala-unstable.vala
+++ b/libsecret/test-vala-unstable.vala
@@ -26,11 +26,11 @@ private static int main (string[] args)
 		MockService.start ("mock-service-normal.py");
 	} catch ( GLib.Error e ) {
 		GLib.error ("Unable to start mock service: %s", e.message);
 	}
 
-	GLib.Test.add_data_func ("/vala/unstable/read-alias", test_read_alias);
+	GLib.Test.add_data_func ("/vala/unstable/read-alias", (GLib.TestDataFunc) test_read_alias);
 
 	var res = GLib.Test.run ();
 	Secret.Service.disconnect ();
 	MockService.stop ();
 	return res;
--- a/libsecret/test-vala-lang.vala
+++ b/libsecret/test-vala-lang.vala
@@ -164,17 +164,17 @@ private static int main (string[] args)
     no_name_schema = new Secret.Schema ("unused.Schema.Name", Secret.SchemaFlags.DONT_MATCH_NAME,
                                         "number", Secret.SchemaAttributeType.INTEGER,
                                         "string", Secret.SchemaAttributeType.STRING);
   }
 
-  GLib.Test.add_data_func ("/vala/lookup/sync", test_lookup_sync);
-  GLib.Test.add_data_func ("/vala/lookup/async", test_lookup_async);
-  GLib.Test.add_data_func ("/vala/lookup/no-name", test_lookup_no_name);
-  GLib.Test.add_data_func ("/vala/store/sync", test_store_sync);
-  GLib.Test.add_data_func ("/vala/store/async", test_store_async);
-  GLib.Test.add_data_func ("/vala/clear/sync", test_clear_sync);
-  GLib.Test.add_data_func ("/vala/clear/async", test_clear_async);
+  GLib.Test.add_data_func ("/vala/lookup/sync", (GLib.TestDataFunc) test_lookup_sync);
+  GLib.Test.add_data_func ("/vala/lookup/async", (GLib.TestDataFunc) test_lookup_async);
+  GLib.Test.add_data_func ("/vala/lookup/no-name", (GLib.TestDataFunc) test_lookup_no_name);
+  GLib.Test.add_data_func ("/vala/store/sync", (GLib.TestDataFunc) test_store_sync);
+  GLib.Test.add_data_func ("/vala/store/async", (GLib.TestDataFunc) test_store_async);
+  GLib.Test.add_data_func ("/vala/clear/sync", (GLib.TestDataFunc) test_clear_sync);
+  GLib.Test.add_data_func ("/vala/clear/async", (GLib.TestDataFunc) test_clear_async);
 
   var res = GLib.Test.run ();
 
   Secret.Service.disconnect ();
   MockService.stop ();
